src/Deck.java

```diff
@@ -0,0 +1,542 @@
+      
+      while (i < singleDeck.cards.length)
+      {
+         System.out.println( singleDeck.cards[i].toString() );
+         i += 1;
+      }
+      System.out.println("----------- 1 Deck (Shuffled) ---------------------");
+      
+      i = 0;
+      singleDeck.init(1);
+      singleDeck.shuffle();
+      
+      while (i < singleDeck.cards.length)
+      {
+         System.out.println( singleDeck.cards[i].toString() );
+         i += 1;
+      }
+      System.out.println("-------------------------------------------------\n");
+      
+   }
+   
+
+   public final int MAX_CARDS = 312;
+   private static Card[] masterPack = new Card[52];
+   private Card[] cards;
+   private int topCard;
+   
+   // Constructors
+   public Deck(int numPacks)
+   {  
+      allocateMasterPack();
+      init(numPacks);
+      topCard = 0;
+   }
+   
+   public Deck()
+   {
+      allocateMasterPack();
+      cards = masterPack;
+   }
+   
+   // Re-populates the cards[] with new cards. Resetting it to its original,
+   //unshuffled state.
+   // Methods
+   public void init(int numPacks)
+   {
+      cards = new Card[numPacks * 52];
+      int index = 0;
+      
+      while (numPacks > 0)
+      {
+         for (int i = 0; i < masterPack.length; i++)
+         {
+            cards[index] = masterPack[i];
+            index++;
+         }
+         
+         numPacks -= 1;
+      }     
+   }
+  
+   // Shuffles the cards[] by iterating through the cards[] and placing the
+   // card in each index in a random index.
+   public void shuffle()
+   {
+      Random rand = new Random();
+      int num;
+      Card tempCard;
+      
+      for (int i = 0; i < cards.length; i++)
+      {
+         num = rand.nextInt(52);
+         tempCard = cards[num];
+         cards[num] = cards[i];
+         cards[i] = tempCard;
+      }
+   }
+   
+   // Returns the card at the 'topCard' position of the deck and simulates the
+   // removal from the deck by incrementing topCard by 1. Also, throws an
+   // exception if the 'topCard' index exceeds the number of cards in the deck.
+   public Card dealCard()
+   {
+      if (topCard == cards.length)
+      {
+         throw new NullPointerException("NO MORE CARDS LEFT TO DEAL!");
+      }
+      
+      int temp = topCard;
+      topCard += 1;
+      return cards[temp];
+   }
+
+   public int getTopCard()
+   {
+      return topCard;
+   }
+
+   // Returns the card in index k of the cards[]. If cards[k] is out of bounds,
+   // then it returns a card with errorFlag set to true.
+   public Card inspectCard(int k)
+   {
+      try
+      {
+         return cards[k];
+      }
+      catch (Exception e)
+      {
+         return new Card('f', Card.Suit.spades);
+      }     
+   }
+   
+   // Populates the masterPack[] with cards.
+   private static void allocateMasterPack()
+   {
+      // Checks to see if masterPack[] has already been filled
+      if (masterPack[0] != null)
+      {
+         return;
+      }
+      
```

Comment:
New packs that are unshuffled are listed 2-10,j-a this would be an ascending chronological order. Set the order of masterpack this was, suit order does not matter, just value